### PR TITLE
Notifications mvp

### PR DESCRIPTION
**Pull request by ateitelbaum**
Added storage and retrieval of notification tokens and tests.

### CODE REVIEW

File: src/main/java/codeu/controller/ChatServlet.java [hunk 1]
```diff
@@ -342,7 +349,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             Instant.now());
 
     //send notification
-    sendNotification.sendMsg(parsedMessageContent);
+    Collection tokens = notificationTokenStore.getAllNotificationTokens();
```
Comment (fatlotus) [inline src/main/java/codeu/controller/ChatServlet.java]: generics?

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 2]
```diff
@@ -41,7 +42,7 @@
   private DatastoreService datastore;
 
   // API Key for Firebase
-  private static String API_ACCESS_KEY="AAAAQ8CY2j0:APA91bHKNT-HNdbxyO-eD671RuQlkZOgMS7VTz66uwtxVr8kDfUDcGCLojY1hgRsXK9IfyE1LmTYarOO6gq_4CThif_6bjgmvh6JIikM28HZTAQs-u7jtEkEUokDHrpWvZ9jsI6Bmfyl";
+  private static String API_ACCESS_KEY="AAAAQ8CY2j0:APA91bGOgQlivDmrkUFqN92TGTuVyooVXFPzggOAjPF79njQDhIE-JcT8UnEUBUJahlv_mcVNj-Yo2N6MTwvZkuIvqIq8vP_kMLT7HlZmyGPcLRLRAYSBfMUgz5D2PVzqXVaYG3Ff5zBmHomQXgAnsh_uqoWZG9Vzw";
```
Comment (fatlotus) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: Is this meant to be secret?

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 3]
```diff
@@ -153,6 +154,36 @@ public PersistentDataStore() {
...
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public Hashtable<UUID, String> loadNotificationTokens() throws PersistentDataStoreException {
+
+    Hashtable<UUID, String> notificationTokens = new Hashtable<>();
+
+    // Retrieve all users from the datastore.
+    Query query = new Query("chat-notification-tokens");
```
Comment (fatlotus) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: Maybe make this a constant, to catch typos at compile time? 

```java
private static final String CHAT_NOTIFICATIONS_TABLE = "chat-notification-tokens";
```

File: src/main/webapp/js/main.js [hunk 4]
```diff
@@ -3,11 +3,15 @@ messaging.usePublicVapidKey("BLIwYdJwu8oCxTYQttElp90iDYg5ooy6tzLvx1g2YznQY2JTIQu
 Notification.requestPermission(function(status) {
     console.log('Notification permission status<:', status);
 });
-
+var xhttp = new XMLHttpRequest();
```
Comment (fatlotus) [inline src/main/webapp/js/main.js]: Optional: if you want to be fancy, you could look at the Fetch API, which is a little cleaner than XHR.

File: src/main/webapp/js/main.js [hunk 5]
```diff
@@ -3,11 +3,15 @@ messaging.usePublicVapidKey("BLIwYdJwu8oCxTYQttElp90iDYg5ooy6tzLvx1g2YznQY2JTIQu
...
 // Get Instance ID token.
 messaging.getToken().then(function(currentToken) {
     if (currentToken) {
-        console.log(messaging.getToken());
+        console.log("Token retrieved");
+        url = "/login?token=" + currentToken+"&t="+Math.random();
```
Comment (fatlotus) [inline src/main/webapp/js/main.js]: I would actually recommend using a POST request here, as they are not cached.

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java (hunk 2 above)
Comment (ateitelbaum) [reply src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: yes. Where can I store it to make it secret?

File: src/main/java/codeu/controller/ChatServlet.java (hunk 1 above)
Comment (ateitelbaum) [reply src/main/java/codeu/controller/ChatServlet.java]: It's there for now until I combine it with the groups functionality.

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java (hunk 2 above)
Comment (ateitelbaum) [reply src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: I put it as an entity in datastore, but it's going to be hard to test.

**Pull request merged**